test_suite/suite_runner.py

```diff
@@ -20,17 +30,23 @@ def compose_testinfra_command(self, output_filepath):
             'py.test',
             ' '.join(test_suite_paths),
             f'--hosts={all_hosts}',
-            f'--ssh-identity-file {self.ssh_identity_file}',
-            f'--junit-xml {output_filepath}'
+            f'--ssh-config {self.ssh_config}',
+            f'--junit-xml {output_filepath}',
+            '--connection=ssh'
         ]
 
+        if self.parallel:
+            command_with_args.append(f'--numprocesses={len(self.instances)}')
+            command_with_args.append(f'--maxprocesses={self.max_processes}')
+
+            # rerun failed tests in case ssh times out or connection is refused by host
+            command_with_args.append('--only-rerun="refused|timeout"')
+            command_with_args.append('--reruns 3')
+            command_with_args.append('--reruns-delay 5')
```

Comment:
I would set `reruns` and `reruns-delay`as variables, so its easier to parametrize and consistent with `max_processes`,